### ISSUE TITLE
Suggestion: Show Videos in Date Order

### ISSUE THREAD

**Issue by tkedwards** [bpatrik/pigallery2#89]
Currently all videos in a folder will appear down the bottom of the page, i.e the videos won't appear in date order. 

I understand videos unfortunately don't have good metadata like photos (EXIF, IPTC etc), but the filesystem modification time can be used instead.

**Comment by bpatrik** [comment]
Hi,

This should actually work. (If not thats a bug). You can try it on the heroku demo. There is a video in the main folder.

Patrik

**Comment by tkedwards** [comment]
Thanks Patrik, I've created a set of 2 test videos and 2 photos and I still see the issue (I'm using the nightly Docker image). The 2 photos are shown first, the 2 videos after. The photos and videos show the time I took them, so you can see they end up out of order.

Download link for the test photos and videos: https://send.firefox.com/download/f3a8398ac3601955/#ZUZByNliLdZRuXrUveprOQ

Screenshot of how they appear in my gallery: 

![grafik](https://user-images.githubusercontent.com/1695095/61676751-686a2180-acfd-11e9-81d9-8f5f36601980.png)

**Comment by bpatrik** [comment]
Hi,

I've looked into it. The issue is that your videos do not contain creation date metadata info.
I [made a fallback](https://github.com/bpatrik/pigallery2/commit/976e50d310478329025858441e1c9c599669068a) to the file creation date. But it is not that reliable (as unzipping your file also overwrite the creation time).

You can check it with the nightly build.

**Comment by tkedwards** [comment]
Thanks, but I wonder if ctime is the right one here? I transfer these files from my phone, either using something like KDE Connect or doing an upload via Nextcloud. In all the process the mtime (Modification Time) is preserved, and this is what Digikam (photo management program on Linux) seems to use. It's also what Google Photos seems to use (the vids are ordered properly there).

Here's a stat on a real example file:

> 

$ stat VID_20190504_173909_resized.mp4 
  File: VID_20190504_173909_resized.mp4
  Size: 33831095        Blocks: 66080      IO Block: 4096   regular file
Device: fd03h/64771d    Inode: 8656406     Links: 1
Access: (0660/-rw-rw----)  Uid: ( 1000/     tim)   Gid: ( 1000/     tim)
Access: 2019-07-15 13:52:06.158163548 +1000
Modify: 2019-05-04 17:40:26.000000000 +1000
Change: 2019-07-23 20:01:02.258859073 +1000
 Birth: 

The mtime (Modify, Modification Time) is the only one that's correct.

**Comment by bpatrik** [comment]
You are right. I assumed ctime means creation time, but it means [change time](https://www.unixtutorial.org/atime-ctime-mtime-in-unix-filesystems).
I changed it to mtime for both videos and photos as a fallback if no metadata present.

**Comment by tkedwards** [comment]
Awesome! Thanks I'll test it out tomorrow (assuming it's in nightly build tonight?)

**Comment by bpatrik** [comment]
Yes. Its already in the nightly build. (Make sure that you have the [latest image](https://hub.docker.com/r/bpatrik/pigallery2/builds). Some said that he did get the latest even after removing the cached images manually)

**Comment by tkedwards** [comment]
Hi, 
Sorry but I haven't had any luck with this. No matter what I do Docker only pulls an old image from 9 days ago. I've tried deleting all bpatrik/pigallery2 with docker rmi, I've tried explicitly running docker pull but no matter what I get the old version:

`
$ docker images
REPOSITORY            TAG                 IMAGE ID            CREATED             SIZE
bpatrik/pigallery2    nightly-stretch     72fe6913921d        9 days ago          1.44GB
`

`
Unable to find image 'bpatrik/pigallery2:nightly-stretch' locally
nightly-stretch: Pulling from bpatrik/pigallery2
a4d8138d0f6b: Pull complete 
dbdc36973392: Pull complete 
f59d6d019dd5: Pull complete 
aaef3e026258: Pull complete 
6e454d3b6c28: Pull complete 
c717a7c205aa: Pull complete 
038b3cfb4595: Pull complete 
a6d4e452109c: Pull complete 
f2e9874fa393: Pull complete 
b569de1480dc: Pull complete 
Digest: sha256:3e51e0d43f319eb3651ab6122cb34cf79aab1896a3db604e6f77216bbd6c9138
Status: Downloaded newer image for bpatrik/pigallery2:nightly-stretch
3a79cb389ab0136c4d7fd64c8087821675501ae61d8ab6ae47a4ba858f8a51d5
`

I've tested it anyway but the same problem (as in the screenshot in my previous comment) is still there

**Comment by bpatrik** [comment]
Hi,

My bad. The caching was enabled for docker hub, so it did not used the latest source for building.
I switched it off.
Should work now.

Best,
Patrik

**Comment by tkedwards** [comment]
Thanks so much for your efforts with this. I got the latest from Docker earlier today, rm'd the Docker container and had it run again, then kicked off a re-index. I haven't had much time today to look at it, but I just checked then and it looks like the videos are in the correct order (in amongst the photos taken at around the same time) in the folder I checked!

Weirdly the folder containing the test photos/vids I gave you hasn't come up in the right order. I'll check more of the 'real' photo folders tomorrow and hopefully they'll all be good
